### PR TITLE
fix: remove duplicate logger for user func call

### PR DESCRIPTION
**Pull request by elbandito**

### CODE REVIEW

File: system/index.js [hunk 1]
```diff
@@ -71,7 +71,10 @@ module.exports = async (message) => {
     }
   }));
 
-  middlewareResult = middlewareResult.concat(logger);
+  // ensure that a logger is added after middleware(s) are processed
+  if (middlewareFns.length > 0) {
```
Comment (jabrown85) [inline system/index.js]: Do we need to add the logger here at all?  If there are no middlwares the logger is passed as the 2nd param...if there are middleware the signature should be 100% controlled by middleware results right?

Comment (elbandito) [reply system/index.js]: yeah, maybe we just have Jing add the logger to the result.  I like that idea.

Comment (elbandito) [reply system/index.js]: @jqian2017 how does this sound to you ^^ ?

Comment (jabrown85) [reply system/index.js]: I thought @jqian2017 didn't even want to add `logger` as another argument. Isn't it part of the `context` being built and not an extra argument?

**Pull request merged**